Automattic/sensei
# Use preloading for initial requests in Importer, Setup Wizard

This uses the `@wordpress/api-fetch` library's functionality to pre-resolve some REST-API requests on the server and send the results along in the initial HTML response. With this, we don't have to show a spinner when opening the importer/exporter/setup wizard while fetching the current state (active step/ongoing job), but can display the correct screen immediately. 

### Changes proposed in this Pull Request

* Add an `apiFetch` middleware that makes sure that preloaded responses are used only once
  * Used by importing `shared/data/api-fetch-preloaded-once`
* Add preloading for getting the active job in importer/exporter
  * Adjust the response of this endpoint to return an empty job instead of `404`, so that it works with preloading
* Add preloading for getting the setup wizard state

### Testing instructions

* Open the Setup Wizard again after finishing it
  * It should display the correct state (all steps completed), without an additional network request and waiting
* Open the Importer, start a new import. Reload the page while it's still in progress.
  * It should show the progress screen immediately
